DBZ-8788 Document mTLS using JKS with Oracle (#6280)
debezium/debezium: documentation/modules/ROOT/pages/connectors/oracle.adoc

When connecting to Oracle using mutual TLS authentication (mTLS), this involves both the connector and the database service proving their identities.
The {prodname} for Oracle connector relies on the Oracle JDBC driver's built-in functionality to support mTLS authentication.

> There are two ways to establish mTLS connections:

Reviewer: ```suggestion
You can use either of the following methods to establish mTLS connections between {prodname} and Oracle:
```

> * xref:#using-mtls-secure-connections-java-key-trust-stores[]

Reviewer: ```suggestion
* xref:#using-mtls-secure-connections-java-key-trust-stores[Using Java key and trust stores to configure mTLS connections].
```

Reviewer: In shared content (community/product), we must provide link text, because the text is not automatically resolved if the link is used downstream.  
In this particular case, the upstream link text will be used, rather than the topic title specified by the downstream Nebel annotation, but better that than no text, and we can live with the small discrepancy as long as the link works.

> * xref:#using-mtls-secure-connections-oracle-wallet[]

Reviewer: ```suggestion
* xref:#using-mtls-secure-connections-oracle-wallet[Using Oracle Wallet to configure mTLS connections].
```

> 

Reviewer: ```suggestion
// Type: procedure
// ModuleID: oracle-connector-using-java-key-trust-stores-to-configure-mtls
// Title: Using Java key and trust stores to configure the Oracle connector to use mTLS
```

[[using-mtls-secure-connections-java-key-trust-stores]]
==== Java key/trust stores

.Prerequisites

> * Verify connector can access key/trust store files directly.
> * Verify database TNS listener supports `tcps` secure connections.

Reviewer: ```suggestion
* Verify that the connector can access the configured key and trust store files.
* Verify that the database TNS (Transparent Network Substrate) listener supports TCPS secure connections.
```

.Procedure

> * Configure the {prodname} Oracle connector, as illustrated below in the example.

Reviewer: ```suggestion
* Configure the {prodname} Oracle connector, as illustrated in the following example. +
```

> 

Reviewer: ```suggestion
+
```

> .Example: {prodname} Oracle connector TLS configuration

Reviewer: ```suggestion
.Example: {prodname} Oracle connector TLS configuration
=====
```

[source,json,indent=0]
----
{
  "database.url": "jdbc:oracle:thin@(DESCRIPTION=(ADDRESS=(PROTOCOL=tcps)(HOST=<host>)(PORT=<port>))(CONNECT_DATA(SERVICE_NAME=<service>)))",
  "driver.javax.net.ssl.keyStore": "<path-to-jks-keystore>",
  "driver.javax.net.ssl.keyStorePassword": "<keystore-password>",
  "driver.javax.net.ssl.keyStoreType": "JKS",
  "driver.javax.net.ssl.trustStore": "<path-to-jks-truststore>",
  "driver.javax.net.ssl.trustStorePassword": "<truststore-password>",
  "driver.javax.net.ssl.trustStoreType": "JKS"
}
----

> 

Reviewer: ```suggestion
=====
+
```
To use example titles, the example must be formatted as an example block.

[NOTE]
====

> It is important for TLS encryption to use `database.url` rather than `database.host`.
> The URL provides a TNS connection string which is what explicitly controls the secure connection requirement by setting the protocol to `tcps`.

Reviewer: ```suggestion
For {prodname} to use TLS encryption to communicate with Oracle, requires a TCPS connection with the server.
To establish a TCPS connection, you must configure the connector to use the `database.url` property, and not the `database.host` property. 
Unlike the `database.host` property, the `database.url` property permits you to define an Oracle TNS (Transparent Network Substrate) connection string that explicitly requires use of the `TCPS` protocol. 
```

====

> 

Reviewer: ```suggestion
// Type: procedure
// ModuleID: oracle-connector-using-oracle-wallet-to-configure-mtls
// Title: Using Oracle wallet to configure the connector to use mTLS
```
